src/index.d.ts

```diff
@@ -0,0 +1,825 @@
+ *
+ * The document is assumed to be a buffer or a string.
+ * Anything else will be converted to a string and then
+ * turned into a buffer.
+ */
+interface TidyBufferStatic {
+    (document: string | Buffer, options: Generated.OptionDict,
+        callback: TidyCallback): void
+
+    (document: string | Buffer, callback: TidyCallback): void
+}
+
+interface TidyOption {
+    readonly name: string
+    readonly type: "boolean" | "integer" | "doctype" | "string"
+    readonly readOnly: boolean
+    readonly pickList: string[]
+    readonly id: number
+    toString(): string
+}
+
+/**
+ * TidyOption type and constructor (not for public use)
+ */
+interface TidyOptionConstructor {
+    new (): TidyOption
+    (): TidyOption
+}
+
+/**
+ * TidyDoc is the central object for dealing with the library at a low level
+ *
+ * Such an object will hold a configuration and be able to process one input
+ * file at a time, while multiple such objects can deal with multiple inputs
+ * simultaneously using an independent configuration for each of them.
+ */
+interface TidyDoc extends Generated.TidyDocOption {
+    // Sync calls
+    cleanAndRepairSync(): string
+    parseBufferSync(document: Buffer): string
+    runDiagnosticsSync(): string
+    saveBufferSync(): Buffer
+    getErrorLog(): string
+
+    // Async calls
+    cleanAndRepair(callback: TidyCallback): void
+    parseBuffer(document: Buffer, callback: TidyCallback): void
+    runDiagnostics(callback: TidyCallback): void
+    saveBuffer(callback: TidyCallback): void
+    tidyBuffer(buf: Buffer, callback: TidyCallback): void
+
+    // batch set/get of options
+    options: OptionDict
+    // Methods that return TidyOption object
+    getOptionList(): TidyOption[]
+    getOption(option: TidyOption): TidyOption
+    getOption(optionId: number): TidyOption
+    getOption(optionName: string): TidyOption
+    /**
+     * FIXME: does this return the list of related options??
+     */
+    optGetDocLinksList(optionId: number): TidyOption[]
+    optGetDocLinksList(optionName: string): TidyOption[]
+    /* FIXME: what does this do? */
+    optGetCurrPick(key: any): any
+}
+
+/**
+ * Constructor
+ * (Can be used with `new` or normal call)
+ */
+interface TidyDocConstructor {
+    new (): TidyDoc
+    (): TidyDoc
+}
+
+/**
+ * Elements of the compat namespace offer compatibility with other
+ * libtidy bindings for node.
+ */
+interface TidyCompat {
+    libtidy: {
+        tidy: {
+            (text: string, callback: (err: any, html: string) => void): void
+            (text: string, options: any, callback: (err: any, html: string) => void): void
+        }
+    }
+}
+
+// just an alias to prevent recursive name resolution
+type _TidyOption = TidyOption
+
+/************************************************************************
+ * The Generated namespace is generated, and not meant for manual edit.
+ *
+ * @see https://gist.github.com/jokester/5c18601db8a9b290fabcfbab4cfc454b
```

Comment:
Better include this script with the codebase. Feel free to create a directory called `util` and put it there. Bonus points if you can rig up recreation of this portion as part of the prepublish phase.